### PR TITLE
ARTEMIS-3657 refactor address docs

### PR DESCRIPTION
**Pull request by jbertram**
Mainly refactoring the address docs. This commit has the following
changes:

 - Remove examples for discouraged use-cases (e.g. using anycast and
multicast on the same address).
 - Reword to use configuration terms wherever possible. For example,
instead of saying "point-to-point" (which is not a configuration term)
say "anycast". References to things like "point-to-point" and
"publish-subscribe" are still there since users are familiar with these
terms. They're just used much less often.
 - Remove duplicate explanation of exclusive queues.
 - Remove duplicate explanation of auto-create and auto-delete elements.
 - Re-create graphics and include the master SVGs for potential updates
later.
 - Give non-destructive queues its own chapter.
 - Add details about specifying routing type using a message property.
 - Update the styling on the user manual's cover page to look better.
 - Lots of re-wording for clarity's sake.
 - Re-order sub-sections for clarity's sake.
 - Break up the address model and the settings documentation. The
settings documentation is large and deserves its own chapter. The
original anchor link is still available with a link to the new chapter.

In general the address-specific documentation should be much more clear,
concise, and consistent now.

### CODE REVIEW

**Comment by jbertram** [comment]
I realize this is a lot to review, but it's been up for over 20 days now with no comments. I optimistically assume that's a good thing. I'll give it a few more days and then merge it if there's no feedback.

File: docs/user-manual/en/address-settings.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+Read more about [slow
+consumers](slow-consumers.md).
+
+`auto-create-jms-queues` is **deprecated**. See `auto-create-queues`. Whether
+or not the broker should automatically create a JMS queue when a JMS message is
+sent to a queue whose name fits the address `match` (remember, a JMS queue is
+just a core queue which has the same address and queue name) or a JMS consumer
+tries to connect to a queue whose name fits the address `match`. Queues which
+are auto-created are durable, non-temporary, and non-transient. Default is
+`true`.
```
Comment (brusdev) [inline docs/user-manual/en/address-settings.md:244]:
```suggestion
`auto-create-jms-queues` is **deprecated**. See `auto-create-queues`.
```
I would remove description for deprecated elements.

File: docs/user-manual/en/address-model.md [hunk 2]
```diff
@@ -1,455 +1,242 @@
...
-      <anycast>
-         <queue name="orders"/>
-      </anycast>
-   </address>
-</addresses>
-```
+> **Note:**
+> 
+> It is possible to define more than one routing type per address, but this
+> typically results in an anti-pattern and is therefore not recommended.
```
[the remark this answers is not in the document]
Comment (jbertram) [reply docs/user-manual/en/address-model.md:47]: I honestly can't think of a use-case for this. Can you? I'd be happy to add an explanation of straight-forward use-case if one exists.

File: docs/user-manual/en/address-model.md [hunk 3]
```diff
@@ -1,455 +1,242 @@
...
-Apache ActiveMQ Artemis supports the ability to filter messages using Apache
-Artemis [Filter Expressions](filter-expressions.md).
+The most common use-case for multicast semantics, sometimes referred to as 
+[publish/subscribe](messaging-concepts.md#publish-subscribe) or "pub/sub",
+involves each application receiving every message sent to an address. Multiple
+applications consuming from a JMS topic is a classic example of this use-case.
+MQTT subscriptions is another supported example of multicast semantics.
 
-Filters can be applied in two places, on a queue and on a consumer.
+In this use-case the broker is configured with an address, `anycast.foo` using
```
[the remark this answers is not in the document]
Comment (jbertram) [reply docs/user-manual/en/address-model.md:140]: Done.

File: docs/user-manual/en/address-model.md [hunk 4]
```diff
@@ -1,455 +1,242 @@
...
+MQTT, STOMP, etc.
+
+## Advanced Manual Configuration
+
+### Fully Qualified Queue Names
+
+In most cases it’s not necessary or desirable to statically configure the
+aforementioned subscription queues. However, there are scenarios where a user
+may want to statically configure a subscription queue and later connect to that
+queue directly using a **Fully Qualified Queue Name** (i.e. FQQN).
```
[the remark this answers is not in the document]
Comment (jbertram) [reply docs/user-manual/en/address-model.md:193]: Done.

File: docs/user-manual/en/address-settings.md [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+
+`auto-delete-jms-queues` is **deprecated**. See `auto-delete-queues`. Whether
+or not the broker should automatically delete auto-created JMS queues when they
+have both 0 consumers and 0 messages. Default is `true`.
```
[the remark this answers is not in the document]
Comment (jbertram) [reply docs/user-manual/en/address-settings.md:248]: Fair enough.

File: docs/user-manual/en/address-model.md [hunk 6]
```diff
@@ -1,455 +1,242 @@
...
 
-In the client code, use both the address name and the queue name when
-requesting a connection from the broker. Remember to use two colons, `::`, to
-separate the names, as in the example Java code below.
+Here's a snippet of Java code using JMS which demonstrates the FQQN syntax:
 
 ```java
-String FQQN = "foo::q1";
+String FQQN = "a1::q1";
 Queue q1 session.createQueue(FQQN);
```
[the remark this answers is not in the document]
Comment (jbertram) [reply docs/user-manual/en/address-model.md:220]: Done.

File: docs/user-manual/en/address-model.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (brusdev) [reply docs/user-manual/en/address-model.md:48]: I have seen more than one routing type per address only as workaround for the following case:
A JMS producer sends messages to `TEST.QUEUE` address while 2 consumers receive messages from 2 FQQN queues `fqqn://TEST.QUEUE::QUEUE1` and `fqqn://TEST.QUEUE::QUEUE2`.
```
<address name="TEST.QUEUE">
    <anycast>
       <queue name="QUEUE1" />
       <queue name="QUEUE2" />
    </anycast>
    <multicast>
       <queue name="TEST.QUEUE" />
    </multicast>
</address>
```

If the multicast queue `TEST.QUEUE` isn't defined the broker will auto-create it as anycast queue distributing messages to `TEST.QUEUE` too.

File: docs/user-manual/en/address-model.md [hunk 7]
```diff
@@ -1,455 +1,242 @@
...
-Figure 1. Point to Point Messaging
-
-#### Using the Anycast Routing Type
-
-Open the file `<broker-instance>/etc/broker.xml` for editing.
-
-Add an address configuration element and its associated queue if they do not
-exist already.
+a single queue on the address.|anycast
+every queue on the address.|multicast
```
Comment (brusdev) [inline docs/user-manual/en/address-model.md:43]:
```suggestion
a single anycast queue on the address.|anycast
every multiple queue on the address.|multicast
```
When the broker routes a message takes into account 2 routing types: the one of the message and the one configured in the queue. I don't know if this is important for users and how to describe it in a clear way. Indeed the suggestion is just a draft to clarify this idea.

File: docs/user-manual/en/address-model.md [hunk 8]
```diff
@@ -1,455 +1,242 @@
...
-   <address name="orders">
-      <anycast>
-         <queue name="orders"/>
-      </anycast>
-   </address>
-</addresses>
-```
+> **Note:**
+> 
+> It is possible to define more than one routing type per address, but this
```
Comment (brusdev) [inline docs/user-manual/en/address-model.md:47]:
```suggestion
> It is possible to define queues with a different routing type for the same address, but this
```

File: docs/user-manual/en/address-model.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (brusdev) [reply docs/user-manual/en/address-model.md:48]: Sounds good

File: docs/user-manual/en/address-model.md [hunk 9]
```diff
@@ -1,455 +1,242 @@
...
+ - **address**
+ - **queue**
+ - **routing type**
 
 ### Address
 
-An address represents a messaging endpoint. Within the configuration, a typical
-address is given a unique name, 0 or more queues, and a routing type.
+Messages are *sent* to an address. An address is given a unique name, a routing
+type, and zero or more queues.
```
[the remark this answers is not in the document]
Comment (jbertram) [reply docs/user-manual/en/address-model.md:22]: I have routing type in here to explain a configuration like this:
```xml
<address name="myTopic">
    <multicast/>
</address>
```
Here the address has a routing type and 0 queues.

File: docs/user-manual/en/address-model.md (hunk 8 above)
Comment (jbertram) [reply docs/user-manual/en/address-model.md:47]: Done.

File: docs/user-manual/en/address-model.md [hunk 10]
```diff
@@ -1,455 +1,242 @@
...
 ### Queue
 
-A queue is associated with an address. There can be multiple queues per
-address. Once an incoming message is matched to an address, the message will be
-sent on to one or more of its queues, depending on the routing type configured.
-Queues can be configured to be automatically created and deleted.
+Messages are *consumed* from a queue. A queue is bound to an address. There can
+be zero or more queues bound to one address. When a message is sent to an
+address it is routed to one or more of its queues based on the configured
+routing type.
```
[the remark this answers is not in the document]
Comment (jbertram) [reply docs/user-manual/en/address-model.md:29]: Done.

**Pull request closed without merging**